lib/beaker/testmode_switcher/beaker_runners.rb

```diff
@@ -96,7 +96,12 @@ def execute_manifest_on(hosts, manifest, opts = {})
                  environment: opts[:environment] || {},
                  acceptable_exit_codes: (0...256))
 
-        handle_puppet_run_returned_exit_code(get_acceptable_puppet_run_exit_codes(opts), res.exit_code)
+        if res.is_a? Array
+          res.each_with_index {|val, index|
+            handle_puppet_run_returned_exit_code(get_acceptable_puppet_run_exit_codes(opts), res["#{index}".to_i].exit_code)}
```

Comment:
`"#{index}".to_i` seems wrong.  Shouldn't it be `val.exit_code`

Also means you don't need the index. so `res.each` should be fine